packages/cli/src/config/settings-manager.ts

```diff
@@ -0,0 +1,110 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as os from 'os';
+import { MCPServerConfig } from '@google/gemini-cli-core';
+
+export type ExtensionScope = 'user' | 'project';
+
+export interface ExtensionMetadata {
+  name: string;
+  source: string;
+  installDate: string;
+  lastUpdated: string;
+  active: boolean;
+  scope: ExtensionScope;
+}
+
+interface SettingsFile {
+  extensions?: Record<string, ExtensionMetadata>;
+  mcpServers?: Record<string, MCPServerConfig>;
+}
+
+export class SettingsManager {
+  private readonly settingsFile: string;
+
+  constructor(scope: ExtensionScope = 'user') {
```

Comment:
Does the settings manager need to be scope aware or can we just pass in the file path in the caller?